src/node.cc

```diff
@@ -5060,13 +5063,26 @@ void deinitV8() {
 
 namespace initialize {
 
-void generateCmdArgsFromProgramName(const std::string& program_name) {
+/**
+ * @brief setCmdArgs builds a continuous adjacent char buffer (argv) from a vector of strings
+ * and stores it in the static cmd_args variable
+ * @param program_name the name of the executable
+ * @param arguments the arguments for the program
+ */
+void setCmdArgs(const std::string& program_name, const std::vector<std::string>& arguments) {
   deinitialize::deleteCmdArgs();
-  int argc = 1;
-  char* program_name_c_string = new char[program_name.length() + 1];
-  std::strcpy(program_name_c_string, program_name.c_str());
-  char** argv = new char*(program_name_c_string);
-  cmd_args = new CmdArgs{argc, argv};
+  cmd_args = new CmdArgs();
+
+  cmd_args->argument_pointer.push_back(const_cast<char*>(cmd_args->argument_data.data() + cmd_args->argument_data.size()));
+  cmd_args->argument_data += program_name;
+  cmd_args->argument_data += char(0x0);
+  for (const auto& s: arguments) {
+    cmd_args->argument_pointer.push_back(const_cast<char*>(cmd_args->argument_data.data() + cmd_args->argument_data.size()));
+    cmd_args->argument_data += s;
+    cmd_args->argument_data += char(0x0);
+  }
+  cmd_args->argc = cmd_args->argument_pointer.size();
+  cmd_args->argv = cmd_args->argument_pointer.data();
```

Comment:
Can we move this into a member function (constructor?) of the `CmdArgs` class?

Reply:
Good point!